src/sql/workbench/browser/modal/modal.ts

```diff
@@ -267,6 +281,19 @@ export abstract class Modal extends Disposable implements IThemable {
 		this.hide();
 	}
 
+	/**
+	 * Used to close modal when a click occurs outside the modal.
+	 * This is exclusive to the Callout.
+	 * @param e The Callout modal click event
+	 */
+	private handleClickOffModal(e: any) {
+		if (e.target.parentElement.closest('.modal-content')) {
```

Comment:
Hm, do we also handle non-click events like keyboard navigation correctly?

Reply:
@Charles-Gagnon In notebooks, if user hits Esc key, this triggers a method which de-activates Edit Mode of the cell. The callout closes, and the cell switches out of edit mode.
In ML extensions, if user hits Esc key, the callout closes. Note: I found on Mac that the user has to hit Esc twice. Not entirely certain what needs to happen in order to have just ONE Esc keypress close it.

Reply:
This should be generic across usage of a callout dialog anywhere - the specific implementations shouldn't really matter I think.

If I have a callout opened what's the general behavior supposed to be? Do you have any examples of other callout dialogs so we can get an idea of how they're used elsewhere? Or some standard you're following here?

Reply:
Handling the click outside of the callout is an existing convention for this type of UI. Specific guidelines can be found here:
https://developer.microsoft.com/en-us/fabric-js/components/callout/callout
I implemented it only for the callout variation of the modal component. Other uses of modal include the flyout variation -- when you click the Connections icon in the Notebook for instance -- these do not follow the same convention of a callout and therefore do not require/need the click handler I use for callouts.